### DISCUSSION TITLE
How to use the keycloak library to implement a user login registration function on the web service enabled by trame？

### DISCUSSION THREAD

**Discussion by liuyichong2250** [Kitware/trame#372] [General]
Hello, I want to use the Trame framework to build a complete web application. And I want to first implement the function of user login and registration. In other discussions, you said that you can use the keycloak library integrated in Trame, but for its use, I don't know much about it yet, and I haven't found any example programs. Can you give some examples?
As a beginner I'm probably asking very superficial questions and I'd really appreciate it if you could answer.

**Comment by jourdain** [comment]
I have yet to complete that module and provide an example. Lucky for you, I have a project that is going to need it soon. So I should be able to tackle it before the end of the year.

**Reply by liuyichong2250** [reply]
Thanks for your answer, I'm looking forward to it.

**Reply by wanqing0421** [reply]
Hi, any progress about the user login features? Can you share some examples?

**Reply by jourdain** [reply]
Sorry the project that was interested in it, changed focus and therefore we could not complete it. If you are interested in funding it, feel free to reach out [here](https://www.kitware.com/trame/) for support.
